src/OSPSuite.SimModel/include/SimModel/EntityWithCachedScaleFactor.h

```diff
@@ -0,0 +1,14 @@
+#ifndef _EntityWithCachedScaleFactor_H_
+#define _EntityWithCachedScaleFactor_H_
+
+namespace SimModelNative
+{
+	class EntityWithCachedScaleFactor
+	{
+	public:
+		//Update cached scale factor of the ODE variable given by <odeIndex> to the new value given by <ODEScaleFactor>
+		virtual void UpdateScaleFactorOfReferencedVariable(const int odeIndex, const double ODEScaleFactor) = 0;
+	};
+
+}
+#endif 
```

Comment:
New interface for all entities which cache a scale factor. Those are not only the formulas, but also FormulaChange (which was forgotten) in the original implementation